lib/charset.c

```diff
@@ -3729,17 +3729,15 @@ static char *qp_encode(const char *data, size_t len, int isheader,
         /* fold header every 78 characters (if possible) */
         size_t i = 0, j = 0, last_wsp = 0;
 
-        while ((len - i > 78) && (j < len)) {
-            j += strcspn(data + j, " \t");
-
+        while ((len - i > 78) && (j += strcspn(data + j, " \t")) < len) {
```

Comment:
A test in charset.testc would be great for this. I think there already are tests for qpencode that could be extended.

Reply:
The line quoted above was actually incorrect, which was discovered with the added cunit tests